sway/desktop/shaders/tex_rgba.frag

```diff
@@ -18,7 +19,9 @@ void main() {
         vec3 target = vec3(dot(irgb, saturation_weight));
         gl_FragColor = vec4(mix(target, irgb, saturation), pixColor.a) * alpha;
     } else {
-        gl_FragColor = texture2D(tex, v_texcoord) * alpha;
+        gl_FragColor = mix(
+            texture2D(tex, v_texcoord), vec4(0.0, 0.0, 0.0, 1.0), dim)
+            * alpha;
```

Comment:
I know it's still a draft but nit picky comment: if you simplify the `vec4(0.0, 0.0, 0.0, 1.0)` to `vec4(vec3(0.0), 1.0)`, the line will be cleaner + the alpha multiplication can fit on the same line